elastic/kibana
# [Onboarding] Remove retrying when fetch index API query fails

## Summary

This Pr set retrying to `false` when fetch Index api returns 404 status. For all other status code, the fetch index api is retried until failure count is 3.

Also updated error component to show error message returned from elasticsearch

### screenshot

**Status code: 404** 

<img width="1286" alt="Screenshot 2024-09-25 at 11 07 20 AM" src="https://github.com/user-attachments/assets/982b5037-b8a5-43d7-bd5f-874929a74ede">

**Status code: 500** 
<img width="1286" alt="Screenshot 2024-09-25 at 11 16 26 AM" src="https://github.com/user-attachments/assets/7b7ecbbe-14af-4d9d-8107-bcee9fa384bf">

**How to test:** 
1. Enable searchIndices plugin in `kibana.dev.yml` as this plugin is behind Feature flag 
```
xpack.searchIndices.enabled: true

```
2. [Create new index](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html)
3. Navigate to `/app/elasticsearch/indices/index_details/${indexName}/data` 
4. [ Delete index](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-delete-index.html)
5. Wait for the error screen to appear. Note error message is shown after one `/internal/index_management/indices/my-index` API is failed

### Checklist

Delete any items that are not applicable to this PR.
